### ISSUE TITLE
custom metric error

### ISSUE THREAD

**Issue by level14taken** [pytorch/xla#1848]
## ❓ Questions and Help
i tried calling the metric(IOU) like this 
```
def train_loop_fn(loader):
    tracker = xm.RateTracker()
    model.train()
    for x, (image, mask) in enumerate(loader): 
      y_pred = model(image)
      loss = loss_fn(y_pred, mask)
      optimizer.zero_grad()
      loss.backward()
      xm.optimizer_step(optimizer)
      IOU= iou(y_pred,mask)
      tracker.add(FLAGS['batch_size'])
      if x % FLAGS['log_steps'] == 0:
        print('[xla:{}]({}) Loss={:.5f} IOU={:.5f}'.format(
          xm.get_ordinal(), x, loss.item(),IOU), flush=True)
    return loss.item(),IOU
```
```
SMOOTH = 1e-6
def iou(outputs: torch.Tensor, labels: torch.Tensor):
    with torch.no_grad():
      outputs = (outputs>0).byte().squeeze(1)  # BATCH x 1 x H x W => BATCH x H x W
      labels= labels.byte()
      intersection = (outputs & labels).float().sum((1, 2))  
      union = (outputs | labels).float().sum((1, 2))     
      iou = (intersection + SMOOTH) / (union + SMOOTH) 
``` 
It resulted in the following error:
`Exception: process 1 terminated with signal SIGSEGV`

I'm using nightly version with 8 num_cores , the error is coming from this line in the Iou method
`intersection = (outputs & labels).float().sum((1, 2)) `

**Comment by dlibenzi** [comment]
Does it happen if you manually create the torch tensors (on XLA device) and pass them to that API?
Trying to get a repro code we can use ...

**Comment by level14taken** [comment]
It seems that i've messed up the mask and image sizes and got that error, nothing to do with TPU
